### DISCUSSION TITLE
How the expiration work?

### DISCUSSION THREAD

**Discussion by AlexZIX** [minio/minio#17669] [Q&A] [answered]
Hi,

I want to use expiration for versioned bucket but can't understand what I'm doing wrong. I've created bucket and enabled versioning. Then I created lifecycle rule "expiry" for a non-current versions and set expiration time to 1 day. Then I upload new file nd remove it. I see that my bucket have 2 versions: NULL VERSION and DELETED. So I expect that in 1 day both of this versions will be removed. But even after 3 days both versions still at my storage.
I've read in docs that objects will be erased not in a flash but when scanner will find they. Is it really long scanning cycle or I doing something wrong? Now only 10TB used in my minio instance.

**Comment by klauspost** [comment, marked as answer]
`mc admin scanner info ALIAS` provides information on scanner speed.

**Reply by AlexZIX** [reply]
Thanks. It means that my configuration right and I just must wait before scanner will remove expired versions? Also how I'll understand that scanner finished his job? There is no versions of expired objects will be like when I just remove it manually?

**Reply by harshavardhana** [reply]
@AlexZIX `mc admin scanner info` says "Last Full Scan" which means it did full scan across all buckets
